src/python/pants/backend/python/target_types.py

```diff
@@ -548,6 +548,25 @@ class PexIncludeToolsField(BoolField):
     )
 
 
+class RunInlineField(BoolField):
+    alias = "run_inline"
+    default = False
+    help = softwrap(
+        """
+        If true, runs of this target with the `run` goal will use the in-repo sources directly.
+        If false, the necessary first-party sources will be copied into a temporary chroot and
+        used from there.
+
```

Comment:
This will not work with codegen, and that should probably be mentioned here. As implemented, it will fail to find those sources (because it will add syspath entries for things which don't exist on disk).

It would be possible to support it if you differentiated which of the sources were generated, and then partitioned between chrooted and un-chrooted syspath entries.

Reply:
Ugh, yeah, that is correct. I should fix that. Maybe we don't actually need to differentiate - we can add the inline and in-sandbox versions of all source roots. Worst case, some of the dirs don't exist.

Reply:
Added support for the codegen case, and a test.